### DISCUSSION TITLE
Error when adding role to new user

### DISCUSSION THREAD

**Discussion by Iainreynard** [chiiya/filament-access-control#50] [Q&A]
Hi - I have implemented the filament-access-control plugin as per the instructions. However, when I create a new user, and what to assign them a role (ie: "staff"), I get the following error:

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'role' in 'field list'
update
  `filament_users`
SET
  `role` = 2,
  `filament_users`.`updated_at` = 2023 -10 -28 14: 04: 32
WHERE
  `id` = 3

I have run all the migrations - not sure what I could be missing. Apologies if its a silly mistake.

Thanks,
Iain

**Comment by yenx2023** [comment]
I have exactly the same problem.
Did you find a solution?

**Comment by Iainreynard** [comment]
Hi

Yes, we ended up writing a roles resource…. please see attached the resource folders for both the User and Roles. Hopefully this assists.





On 22 Nov 2023, at 13:21, yenx2023 ***@***.***> wrote:



I have exactly the same problem.
Did you find a solution?

**Comment by yenx2023** [comment]
thanks you for your response.

the problem apparently is disabling model guard in AppServiceProvider
